### PR TITLE
WIP : Indicate to users how "trending" works

### PR DESCRIPTION
**Pull request by auberanger**
This PR is related to the following issue #1341 
I've just give a try to a simple help icon with a tooltip in order to explain how trending works. It looks like this : 
screenshot_2018-12-05 trending videos - peertube
screenshot_2018-12-05 trending videos - peertube-clicked

In my view, this solution is not really satisfying. I was thinking about something subtler, as I'm going to explain in the issue comments.

### CODE REVIEW

**Comment by Chocobozzz** [comment]
I would prefer to use a tooltip as you said in https://github.com/Chocobozzz/PeerTube/issues/1341#issuecomment-444502177

https://ng-bootstrap.github.io/#/components/tooltip/examples

You could change the title to "Trending videos in the last x days", too

**Comment by auberanger** [comment]
Following your advice, I've changed the help component by a ng-tooltip and the title of the trending page. However, I don't manage to find where exactly is defined the Trending algorithm. It seems to be related to the videoService.getVideos() method and to the RestService class but I don't understand precisely how. I've also looked into the admin configuration as suggested in this [comment](https://github.com/Chocobozzz/PeerTube/issues/1207#issuecomment-427561081), without success.

Also, I don't manage to find any mention of a parameter for the last "X" days.

Would you have a tip for helping me about this ?

**Comment by Chocobozzz** [comment]
Sorry indeed I have not been very clear

> Also, I don't manage to find any mention of a parameter for the last "X" days.

You'll have to add it to [the config endpoint](https://github.com/Chocobozzz/PeerTube/blob/develop/server/controllers/api/config.ts#L54) from [the constants](https://github.com/Chocobozzz/PeerTube/blob/develop/server/initializers/constants.ts#L220)

The trending algorithm is very simple: it sum the views made in the last x days, and sort them. (see the [query builder](https://github.com/Chocobozzz/PeerTube/blob/develop/server/models/video/video.ts#L1345) and [the sort](https://github.com/Chocobozzz/PeerTube/blob/develop/server/models/utils.ts#L22)

**Comment by auberanger** [comment]
Hello @Chocobozzz, you can find a first draft of the resolution of the issue in the previous commits.

File: client/src/app/shared/video/abstract-video-list.html [hunk 1]
```diff
@@ -1,8 +1,18 @@
 <div [ngClass]="{ 'margin-content': marginContent }">
   <div class="videos-header">
     <div *ngIf="titlePage" class="title-page title-page-single">
-      {{ titlePage }}
+      <ng-template i18n #tipContent>
+        The <b>Trending algorithm</b> sums the views made in the last {{ trendingDays }} days for the videos of the related instances, and sort them.
```
Comment (Chocobozzz) [inline client/src/app/shared/video/abstract-video-list.html]: I think we should remove the `<b>` tag, so we can put this text in a `titleTooltip` variable in `abstract-video-list.ts` (the children choose to fill it or not). This way, you would not have this ngIf in the template:

```
<div placement="bottom" [ngbTooltip]="titleTooltip" container="body">
  {{ titlePage }}
</div>
```

In the trending videos component, `titlePage` should be `titlePage = this.i18n('{{titlePage}} for the last {{trendingDays}} days', ...)`

Comment (auberanger) [reply client/src/app/shared/video/abstract-video-list.html]: Thanks for this much more elegant solution !

**Comment by Aluriak** [comment]
IMHO,

    The Trending algorithm sums the views made in the last {{ trendingDays }} days for the videos of the related instances, and sort them

is quite technical and would be simpler for most people if worded as

    trending videos are those totalizing the greatest number of views during the last {{ trendingDays }} days

And also, maybe use *during the last day* instead for when `trendingDays` is 1.

**Comment by auberanger** [comment]
I agree with you @Aluriak. Howewer, would it not be even clearer to use _during the last 24 hours_ instead of _during the last day_ instead for when `trendingDays` is 1 ?

**Comment by Aluriak** [comment]
> would it not be even clearer to use *during the last 24 hours instead of during the last day*

yes, it would.

**Comment by Chocobozzz** [comment]
Thanks @auberanger

**Pull request merged**